### PR TITLE
Long subject names

### LINKED ISSUE #468: Feature Request — Long subject names
**Job**
Add long subject names (e.g. CSCI - Computer Science) to the Explore page

**Context**
It's a lot easier to read a subject name rather than the associated code, especially for incoming freshmen.

### PR DESCRIPTION
**Pull request by i-am-david-liu**
**Issue**
Closes #468.

**Additional Info**
New dev here. To solve this issue, the only solution I was able to implement was hardcoding subject code <-> subject name associations in `utils.js`. I'm not satisfied with this, but since the change was primarily visual and a one-off, I decided to make a request.

Before this, I tried inserting the long subject names into the .csv, but this was difficult and I wasn't able to figure it out. If any experienced devs can point me in the right direction, that would be helpful.

**Screenshots**

### CODE REVIEW

**Comment by marchdan** [comment]
Can you include screenshots of the result (light mode and dark mode) in the PR description?

**Review by antmcgrann — APPROVED** [review]

File: src/web/src/utils.js [hunk 1]
```diff
@@ -380,3 +380,24 @@ export const exportScheduleToImage = (
       console.log("Oh No, Something Went Wrong!", error);
     });
 };
+
+export const getLongName = (department) => {
+  var dict = {
```
Comment (marchdan) [inline src/web/src/utils.js:385]: STSO= Science & Technology Studies

File: src/web/src/pages/CourseExplorer.vue [hunk 2]
```diff
@@ -122,7 +122,7 @@ export default {
         if (!schoolsMajorDict[c.school]) {
           schoolsMajorDict[c.school] = new Set();
         }
-        schoolsMajorDict[c.school].add(c.department);
+        schoolsMajorDict[c.school].add(c.department);
```
Comment (marchdan) [inline src/web/src/pages/CourseExplorer.vue:125]: No need for this file to be in the pr. Remove the space for a more clean PR.

**Review by marchdan — CHANGES_REQUESTED** [review]
I don't think package-lock.json and package.json need to be included in your PR, has caused issues in the past.

Also, if you know why "YACS" in the upper left hand corner is too big and bold on this PR, please lmk...

File: src/web/src/utils.js (hunk 1 above)
Comment (i-am-david-liu) [reply src/web/src/utils.js:385]: This subject code doesn't exist in `school-department-mapping.yaml`, should the change be reflected there too? I'm assuming it replaces STSH and STSS

**Comment by i-am-david-liu** [comment]
The bold YACS logo is probably from https://github.com/YACS-RCOS/yacs.n/pull/443. How do I uncheck a commit from a PR?

File: src/web/src/utils.js (hunk 1 above)
Comment (marchdan) [reply src/web/src/utils.js:385]: > This subject code doesn't exist in `school-department-mapping.yaml`, should the change be reflected there too? I'm assuming it replaces STSH and STSS

That is correct, and it should fall under HASS.

**Comment by marchdan** [comment]
> The bold YACS logo is probably from #443. How do I uncheck a commit from a PR?

If the change was made in a previously merged PR it needs a new issue and PR to fix it. I can look at the old PR and issue and see if there is a reason it was changed.

**Comment by marchdan** [comment]
@i-am-david-liu It looks like all of this information is already in rpi_data/school-department-mapping.yaml. You should be able to utilize that to put the department's long name into the CSV and parse it through from there. Thoughts?

**Comment by i-am-david-liu** [comment]
Yes, that's the current plan. I'm actually going to put this branch on hold for now as I work on #381, which I'll make a pull request later today. 

Longer answer: While working on #381 I finally understood how the .csv was generated and parsed, but that knowledge wasn't available to me when I created this issue. Since both issues parallel each other (add to .csv -> parse from .csv -> display on page), I'm hoping that by finishing #381 first, I should be able to knock this one out pretty easily.

**Comment by marchdan** [comment]
> Yes, that's the current plan. I'm actually going to put this branch on hold for now as I work on #381, which I'll make a pull request later today.
> 
> Longer answer: While working on #381 I finally understood how the .csv was generated and parsed, but that knowledge wasn't available to me when I created this issue. Since both issues parallel each other (add to .csv -> parse from .csv -> display on page), I'm hoping that by finishing #381 first, I should be able to knock this one out pretty easily.

Sounds good! I am the primary developer for the csv logic, so feel free to reach out on discord with any questions you have.

**Comment by DarknessUnder** [comment]
The code has been approved and merged back to master on the different pull request.

**Pull request closed without merging**